src/flextopo.jl

```diff
@@ -0,0 +1,795 @@
+        flextopo.qpercolation[i] = setindex(flextopo.qpercolation[i], qpercolation, k)
+        flextopo.rootevap[i] = setindex(flextopo.rootevap[i], rootevap, k)
+        flextopo.actevap[i] = setindex(
+            flextopo.actevap[i],
+            flextopo.rootevap[i][k] + flextopo.intevap[i][k] + flextopo.hortonevap[i][k],
+            k,
+        )
+        flextopo.wb_rootzone[i] = setindex(flextopo.wb_rootzone[i], wb_rootzone, k)
+
+        #average storage over classes
+        flextopo.srootzone_m[i] = sum(flextopo.rootzonestorage[i] .* flextopo.hrufrac[i])
+        flextopo.srootzone_over_srmax_m[i] = sum(flextopo.srootzone_over_srmax[i] .* flextopo.hrufrac[i])
+        flextopo.rootevap_m[i] = sum(flextopo.rootevap[i] .* flextopo.hrufrac[i])
+        flextopo.actevap_m[i] = sum(flextopo.actevap[i] .* flextopo.hrufrac[i])
+    end
+end
+
+function rootzone_storage(flextopo::FLEXTOPO, config)
+    k = flextopo.kclass[1]
+    for i = 1:flextopo.n
+        # rootzonestorage = flextopo.rootzonestorage[i] + qhortonrootzone > flextopo.srmax[i] ? flextopo.srmax[i] : flextopo.rootzonestorage[i] + qhortonrootzone
+
+        #added water to the root-zone. NB: if no horton storages: qhortonrootzone is in fact Pe!! (effective precip). 
+        rootzonestorage = flextopo.rootzonestorage[i][k] + flextopo.qhortonrootzone[i][k]
+        # if soil is filled until max capacity, additional water runs of directly
+        directrunoff = max(rootzonestorage - flextopo.srmax[i][k], 0)
+        #update rootzonestorage
+        rootzonestorage = rootzonestorage - directrunoff
+        #net water which infiltrates in root-zone
+        netin_rootzone = flextopo.qhortonrootzone[i][k] - directrunoff
+
+        #evaporation from the rootzone
+        rootevap =
+            rootzonestorage > flextopo.srmax[i][k] * flextopo.lp[i][k] ?
+            min(rootzonestorage, flextopo.potsoilevap[i][k]) :
+            min(
+                flextopo.potsoilevap[i][k] *
+                (rootzonestorage / (flextopo.srmax[i][k] * flextopo.lp[i][k])),
+            )
+        #update storage
+        rootzonestorage = rootzonestorage - rootevap
+
+        #excess water from beta function of netin_rootzone
+        qrootzone_in = netin_rootzone * (1 - pow(1 - rootzonestorage / flextopo.srmax[i][k], flextopo.beta[i][k]))
+        #update storage
+        rootzonestorage = rootzonestorage - qrootzone_in
+
+        #total water out of root-zone consists of directrunoff and excess water from beta function
+        qrootzone = directrunoff + qrootzone_in
+
+        #percolation 
+        qpercolation = flextopo.perc[i][k] * rootzonestorage / flextopo.srmax[i][k]
+        rootzonestorage = rootzonestorage - qpercolation
+
+        #TODO? eventueel niet een volgorde voor de verschillende processen opleggen - eerst rootevap, dan qrootzone, dan Perc, maar alles naar rato laten gebeuren zoals in code python. 
```

Comment:
Que? Let's stick to English comments ;)

Reply:
haha, it is removed now in commit 7dde6ff6469d2ec5e4b975f9dd0f455987dff313